MVP Module/Source/Feed UI/Controllers/FeedViewController.swift

```
@@ -24,19 +24,31 @@ public final class FeedViewController: UITableViewController, UITableViewDataSou
 	@IBAction private func refresh() {
 		delegate?.didRequestFeedRefresh()
 	}
+    
+    @IBOutlet public var errorView: ErrorView?
 	
 	func display(_ viewModel: FeedLoadingViewModel) {
+        errorView?.hideMessage()
 		if viewModel.isLoading {
 			refreshControl?.beginRefreshing()
-		} else {
-			refreshControl?.endRefreshing()
-		}
-	}
-	
-	func display(_ cellControllers: [FeedImageCellController]) {
-		tableModel = cellControllers
+        } else {
+            showErrorMessage(message: viewModel.error)
+        }
 	}
-	
+    
+  private func showErrorMessage(message: String?) {
+      if let message = message {
+          errorView?.show(message: message)
+      } else {
+          errorView?.hideMessage()
+      }
+      refreshControl?.endRefreshing()
+  }
```

Showing an error and showing a loading indicator should be separate concerns.

For example, it could show an error while loading - and it could *not* show an error while not loading. And so on.

The presentation logic is already defined by the `Presenter`, so the UI shouldn't decide when to show/hide the error depending on the loading state with the chained if/else statements. 

It'd be best to render the views based on the data received. "If there's an error string, show the error regardless of the loading state." "If isLoading is true, show the loading indicator regardless of the error message state." And so on.

This way, the views and controllers are much easier to implement and understand. No need for chained if/else.